src/jit/alloc.cpp

```diff
@@ -8,10 +8,38 @@
 #pragma hdrstop
 #endif // defined(_MSC_VER)
 
+//------------------------------------------------------------------------
+// PooledAllocator:
+//    This subclass of `ArenaAllocator` is a singleton that always keeps
+//    a single default-sized page allocated. We try to use the singleton
+//    allocator as often as possible (i.e. for all non-concurrent
+//    method compilations).
+class PooledAllocator : public ArenaAllocator
+{
+private:
+    enum
+    {
+        POOLED_ALLOCATOR_NOTINITIALIZED = 0,
+        POOLED_ALLOCATOR_IN_USE = 1,
+        POOLED_ALLOCATOR_AVAILABLE = 2,
+        POOLED_ALLOCATOR_SHUTDOWN = 3,
+    };
```

Comment:
I've been wondering if we should have an *initializing* state -- seems that it might be cleaner than the "failure to get the first page" fall back below. Means two ICE's along the init path but probably not that big of a deal since it only happens once per process lifetime.